src/policy/concrete.rs

```diff
@@ -100,6 +100,20 @@ pub enum PolicyError {
     DuplicatePubKeys,
 }
 
+/// Descriptor context for [`Policy`] compilation into a [`Descriptor`]
+pub enum DescriptorCtx<Pk> {
+    /// [Bare][`Descriptor::Bare`]
+    Bare,
+    /// [Sh][`Descriptor::Sh`]
+    Sh,
+    /// [Wsh][`Descriptor::Wsh`]
+    Wsh,
+    /// Sh-wrapped [Wsh][`Descriptor::Wsh`]
+    ShWsh,
+    /// [Tr][`Descriptor::Tr`], where the Option<Pk> corresponds to the internal_key.
```

Comment:
the documentation here should highlight that his is used as the internal key only when there is no inferred from Policy